src/main/java/reiff/finalproject/ImageFrame.java

```diff
@@ -0,0 +1,35 @@
+package reiff.finalproject;
+
+import javax.swing.*;
+import java.awt.*;
+import java.net.URL;
+
+public class ImageFrame extends JFrame {
+    public ImageFrame(String imageUrl) {
+        setTitle("Art Image");
+        setSize(800, 600);
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        try {
+            ImageIcon imageIcon = new ImageIcon(new URL(imageUrl));
+            Image image = imageIcon.getImage();
+            int width = image.getWidth(null);
+            int height = image.getHeight(null);
+
+            if (width > 800) {
+                int newHeight = (int) ((800.0 / width) * height);
+                image = image.getScaledInstance(800, newHeight, Image.SCALE_SMOOTH);
```

Comment:
You can pass -1 for newHeight and the method will do the proportional calculation itself.

I actually don't understand why you are doing it this way here where you do it correctly in RijksSearchFrame